metadata-io/src/main/java/com/linkedin/metadata/search/elasticsearch/query/ESBrowseDAO.java

```diff
@@ -74,14 +76,26 @@ public BrowseResult browse(@Nonnull String entityName, @Nonnull String path, @Nu
 
     try {
       final String indexName = indexConvention.getIndexName(entityRegistry.getEntitySpec(entityName));
+
       final SearchResponse groupsResponse =
-          client.search(constructGroupsSearchRequest(indexName, path, requestMap, 0, 1000), RequestOptions.DEFAULT);
+          client.search(constructGroupsSearchRequest(indexName, path, requestMap), RequestOptions.DEFAULT);
+      final BrowseResultMetadata browseResultMetadata = extractGroupsResponse(groupsResponse, path, from, size);
+      final int numGroups = browseResultMetadata.getTotalNumEntities().intValue();
+
+      // Based on the number of groups returned, compute the from and size to query for entities
+      int entityFrom = Math.max(from - numGroups, 0);
+      int entitySize = Math.min(Math.max(from + size - numGroups, 0), size);
       final SearchResponse entitiesResponse =
-          client.search(constructEntitiesSearchRequest(indexName, path, requestMap, from, size),
+          client.search(constructEntitiesSearchRequest(indexName, path, requestMap, entityFrom, entitySize),
               RequestOptions.DEFAULT);
-      final BrowseResult result = extractQueryResult(groupsResponse, entitiesResponse, path, from);
-      result.getMetadata().setPath(path);
-      return result;
+      final int numEntities = (int) entitiesResponse.getHits().getTotalHits().value;
+      final List<BrowseResultEntity> browseResultEntityList = extractEntitiesResponse(entitiesResponse, path);
+
+      return new BrowseResult().setMetadata(browseResultMetadata)
+          .setEntities(new BrowseResultEntityArray(browseResultEntityList))
+          .setNumEntities(numGroups + numEntities)
```

Comment:
should this be numElements or total or some other name that indicates it is a sum of entities + groups?

I would expect numEntities to be numEntities + group.map(group -> group.count).sum()

Reply:
Yeah the naming here is confusing... I'd agree I'd think num entities would be either 

a. The number of concrete entities matching the exact path OR 
b. The total number of entities somewhere under this path

In fact, it may be useful to include both of these in the response separately. 

Can we nip this in the bud with this PR?

Reply:
Unfortunately BrowseResult PDL is in datahub-gms right now. We can make the above changes after moving them over here.